### DISCUSSION TITLE
Canonical way to run actions when a *specific package* is upgraded?

### DISCUSSION THREAD

**Discussion by ghost** [Homebrew/discussions#5374] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.0
ORIGIN: https://github.com/Homebrew/brew
HEAD: 8378cc825d83acffd125fb0fec041793df378a57
Last commit: 3 days ago
Core tap JSON: 17 May 01:54 UTC
Core cask tap JSON: 17 May 01:54 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: Eleanors-Laptop.local
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.
4/bin/ruby
CPU: octa-core 64-bit arm_ibiza
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.4.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

On my system I have opam and DOOM Emacs, and as I get more into things I see a significant possibility of more domain-specific package managers. I think it'd be overkill to `opam update && opam upgrade -y $(opam list -s --roots)` and `.config/emacs/bin/doom upgrade` every day, but certainly I want to do it every time `opam` resp. `emacs-plus` gets upgraded. My question is, how would I go about doing that? I know the core team is averse to adding a broad `brew upgrade` hook, and I understand that; but this is much more specific, and I don't see a clear way to do it with the available tools. I suppose I could learn Raku and write a script that splits and scans the output to figure out which packages were upgraded (and doesn't blow up on any of the myriad other messages the command might generate); but as I'm sure you understand I'd _really_ rather not do that.

**Comment by gromgit** [comment]
If I understand you correctly, a simple `brew upgrade <pkg1> <pkg2>...` upgrades only the packages you specify _and_ all their dependencies. Stick that in your crontab to run it regularly; if there aren't any upgrades, nothing happens.

**Reply by ghost** [reply]
No, that is not what I mean. I've updated the question for clarity.

**Reply by gromgit** [reply]
> I want to do it every time `opam` resp. `emacs-plus` gets upgraded.

What does "resp." mean? And how is Homebrew involved when it looks like you're trying to monitor `opam` package DB updates?

**Reply by ghost** [reply]
"resp." means "respectively". Homebrew is involved because I want to check for opam DB updates _precisely when opam itself is upgraded by Homebrew_.

**Reply by carlocab** [reply]
There isn't really a "canonical" way to do this. I suppose you could set up `watchman` to check whenever `opam` changes and then get it to do what it is you want to do when that happens.

**Comment by ghost** [comment]
Well, that sucks. I can see a giant headache pursuing that approach. Thanks anyway though.

I guess it is best practice to do it manually every time. It’s not like these are security-critical apps, and there might be advisories I should see.
